app/WebParsing/PostParser.hs

```diff
@@ -59,16 +62,32 @@ categoryParser tags fullPostName firstCourse liPartitions = do
             liftIO $ print failedString
             return ()
     where
-        parsed = case liPartitions of
+        parsed = case listPartitions of
             [] -> P.parse (generalCategoryParser fullPostName firstCourse) failedString (innerText tags)
             partitionResults -> do
-                let categories = map parseLi partitionResults
                 post <- P.parse (postInfoParser fullPostName firstCourse) failedString (innerText tags)
-                return (post, categories)
+                return (post, partitionResults)
 
 parseLi :: [Tag T.Text] -> T.Text
 parseLi liPartition = do
     let parsed = P.parse parseCategory failedString (innerText liPartition)
     case parsed of
         Right category -> category
         Left _ -> ""
+
+getNumberedPartitions :: [Tag T.Text] -> [[Tag T.Text]]
+getNumberedPartitions tags = do
+    let pTags = partitions isPTag tags
+        pTagsWithoutBr = map (splitWhen isBRTag) pTags
+    concat pTagsWithoutBr
+    where
+        isPTag tag = isTagOpenName "p" tag
```

Comment:
Don't define separate functions; inline them. Note that `isPTag = isTagOpenName "p"`.